src/index/blockfilterindex.h

```diff
@@ -30,6 +33,9 @@ class BlockFilterIndex final : public BaseIndex
     bool ReadFilterFromDisk(const FlatFilePos& pos, BlockFilter& filter) const;
     size_t WriteFilterToDisk(FlatFilePos& pos, const BlockFilter& filter);
 
+    RecursiveMutex m_cs_headers_cache;
```

Comment:
```suggestion
    Mutex m_cs_headers_cache;
```

nit (feel free to ignore). I haven't tested this, but from reading the code, a non-recursive mutex should be sufficient.

Reply:
Agree. I'll change to `Mutex`.

Reply:
Done